p2p/libp2p/netMessenger_test.go

```diff
@@ -1076,10 +1085,90 @@ func generateConnWithRemotePeer(pid core.PeerID) network.Conn {
 	}
 }
 
-func TestLibp2pMessenger_SendDirectWithMockNetToConnectedPeerShouldWork(t *testing.T) {
+func TestLibp2pMessenger_SendDirectWithRealMessengersShouldWork(t *testing.T) {
 	msg := []byte("test message")
 
-	_, messenger1, messenger2 := createMockNetworkOf2()
+	args := libp2p.ArgsNetworkMessenger{
+		Marshalizer:   &testscommon.ProtoMarshalizerMock{},
+		ListenAddress: libp2p.ListenLocalhostAddrWithIp4AndTcp,
+		P2pConfig: config.P2PConfig{
+			Node: config.NodeConfig{
+				Port:                  "0",
+				ConnectionWatcherType: "print",
+			},
+			KadDhtPeerDiscovery: config.KadDhtPeerDiscoveryConfig{
+				Enabled: false,
+			},
+			Sharding: config.ShardingConfig{
+				Type: p2p.NilListSharder,
+			},
+		},
+		SyncTimer:            &libp2p.LocalSyncTimer{},
+		PreferredPeersHolder: &p2pmocks.PeersHolderStub{},
+		PeersRatingHandler:   &p2pmocks.PeersRatingHandlerStub{},
+	}
+	messenger1, _ := libp2p.NewNetworkMessenger(args)
+	messenger2, _ := libp2p.NewNetworkMessenger(args)
+
+	adr2 := messenger2.Addresses()[0]
+
+	fmt.Printf("Connecting to %s...\n", adr2)
+
+	_ = messenger1.ConnectToPeer(adr2)
+
+	wg := &sync.WaitGroup{}
+	chanDone := make(chan bool)
+	wg.Add(1)
+
+	go func() {
+		wg.Wait()
+		chanDone <- true
+	}()
+
+	prepareMessengerForMatchDataReceive(messenger2, msg, wg)
```

Comment:
can you make it clear in the test that real sig verification is done?

Reply:
added signature size checks on the refactored & new test